sdk/python/kfp/v2/dsl/component_bridge.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Function for creating ContainerOp instances from component spec."""
+
+import copy
+from typing import Any, Mapping
+
+from kfp import dsl
+from kfp.components import structures
+from kfp.components._components import _default_component_name
+from kfp.components._components import _resolve_command_line_and_paths
+from kfp.components._naming import _sanitize_python_function_name
+from kfp.components._naming import generate_unique_name_conversion_table
+from kfp.dsl import types
+from kfp.v2.dsl import type_utils
+from kfp.v2.proto import pipeline_spec_pb2
+
+
+def _create_container_op_from_component_and_arguments(
+    component_spec: structures.ComponentSpec,
+    arguments: Mapping[str, Any],
+    component_ref: structures.ComponentReference = None,
+) -> dsl.ContainerOp:
+
+  pipeline_task_spec = pipeline_spec_pb2.PipelineTaskSpec()
+  pipeline_task_spec.task_info.name = component_spec.name
+  # might need to append suffix to exuector_label to ensure its uniqueness?
+  pipeline_task_spec.executor_label = component_spec.name
+
+  # Check types of the reference arguments and serialize PipelineParams
+  arguments = arguments.copy()
+  for input_name, argument_value in arguments.items():
+    if isinstance(argument_value, dsl.ContainerOp):
```

Comment:
I think we can remove this condition now?

Reply:
I'm not sure about that.

Reply:
The code below only explicitly allowlists `PipelineParam` and primitive typed arguments, thus `ContainerOp` will directly fallback to the `else` clause, which is an error.

Reply:
Moved this condition down below, but keep it separate from the else to retain the `TypeError` instead of `NotImplementedError`.